test/development/app-dir/stitching-errors/app/browser/caught/page.js

```diff
@@ -0,0 +1,44 @@
+'use client'
+
+import { Component } from 'react'
+
+class MyErrorBoundary extends Component {
+  static getDerivedStateFromError(error) {
+    return { error }
+  }
+
+  state = { error: null }
+
+  render() {
+    if (this.state.error) {
+      return 'failed'
+    }
+    return this.props.children
+  }
+}
+
+function Inner() {
+  return (
+    <MyErrorBoundary>
+      <Thrower />
+    </MyErrorBoundary>
+  )
+}
+
+function Thrower() {
+  useErrorHook()
+}
+
+function useThrowError() {
+  if (typeof window !== 'undefined') {
+    throw new Error('browser error')
+  }
```

Comment:
Why not error consistently on the Client? This is technically a hydration mismatch.